commands/cdn/distribution/list.go

```diff
@@ -0,0 +1,92 @@
+package distribution
+
+import (
+	"context"
+	"fmt"
+	"github.com/ionos-cloud/ionosctl/v6/commands/cdn/completer"
+	"github.com/ionos-cloud/ionosctl/v6/internal/constants"
+	"github.com/ionos-cloud/ionosctl/v6/internal/core"
+	"github.com/ionos-cloud/ionosctl/v6/internal/printer/json2table"
+	"github.com/ionos-cloud/ionosctl/v6/internal/printer/json2table/jsonpaths"
+	"github.com/ionos-cloud/ionosctl/v6/internal/printer/jsontabwriter"
+	"github.com/ionos-cloud/ionosctl/v6/internal/printer/tabheaders"
+	cdn "github.com/ionos-cloud/sdk-go-cdn"
+	"github.com/spf13/viper"
+)
+
+func List() *core.Command {
+	cmd := core.NewCommand(
+		context.Background(), nil, core.CommandBuilder{
+			Namespace: "cdn",
+			Resource:  "distribution",
+			Verb:      "list",
+			Aliases:   []string{"ls"},
+			ShortDesc: "Retrieve all distributions using pagination and optional filters",
+			Example:   `ionosctl cdn ds list`,
+			PreCmdRun: func(c *core.PreCommandConfig) error {
+				return nil
+			},
+			CmdRun: func(c *core.CommandConfig) error {
+				return listDistributions(c)
+			},
+			InitClient: true,
+		},
+	)
+
+	cmd.AddStringFlag(constants.FlagCDNDistributionFilterDomain, "", "", "Filter used to fetch only the records that contain specified domain.")
+	cmd.AddSetFlag(constants.FlagCDNDistributionFilterState, "", "", []string{"AVAILABLE", "BUSY", "FAILED", "UNKNOWN"}, "Filter used to fetch only the records that contain specified state.")
+	cmd.AddInt32Flag(constants.FlagOffset, "", 0, "The first element (of the total list of elements) to include in the response. Use together with limit for pagination")
+	cmd.AddInt32Flag(constants.FlagLimit, "", 0, constants.DescMaxResults)
+
+	cmd.Command.PersistentFlags().StringSlice(
+		constants.ArgCols, nil,
+		fmt.Sprintf(
+			"Set of columns to be printed on output \nAvailable columns: %v",
+			allCols,
+		),
+	)
+
+	return cmd
+}
+
+func listDistributions(c *core.CommandConfig) error {
+	ls, err := completer.Distributions(
+		func(req cdn.ApiDistributionsGetRequest) (cdn.ApiDistributionsGetRequest, error) {
+			if fn := core.GetFlagName(c.NS, constants.FlagCDNDistributionFilterState); viper.IsSet(fn) {
+				req = req.FilterState(viper.GetString(fn))
+			}
+			if fn := core.GetFlagName(c.NS, constants.FlagCDNDistributionFilterDomain); viper.IsSet(fn) {
+				req = req.FilterDomain(viper.GetString(fn))
+			}
+			if fn := core.GetFlagName(c.NS, constants.FlagOffset); viper.IsSet(fn) {
+				req = req.Offset(viper.GetInt32(fn))
+			}
+			if fn := core.GetFlagName(c.NS, constants.FlagLimit); viper.IsSet(fn) {
+				req = req.Limit(viper.GetInt32(fn))
```

Comment:
I know it would make sense to use `--limit` but sadly we already use `--max-results` for all other products, and changing the name for CDN will be confusing for the user

Perhaps in the future we could come up with a better pagination system